src/domain/dataset-update-flow/src/aggregates/update_schedule.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use event_sourcing::*;
+use opendatafabric::DatasetID;
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Aggregate, Debug)]
+pub struct UpdateSchedule(Aggregate<UpdateScheduleState, (dyn UpdateScheduleEventStore + 'static)>);
+
+impl UpdateSchedule {
+    /// Creates a dataset update flow
+    pub fn new(now: DateTime<Utc>, dataset_id: DatasetID, schedule: Schedule) -> Self {
+        Self(
+            Aggregate::new(
+                dataset_id.clone(),
+                UpdateScheduleEventCreated {
+                    event_time: now,
+                    dataset_id,
+                    schedule,
+                },
+            )
+            .unwrap(),
+        )
+    }
+
+    /// Pause flow
+    pub fn pause(
+        &mut self,
+        now: DateTime<Utc>,
```

Comment:
It seems unfortunate that `now` time appears in every single method now. Should we make time source a filed in the `Aggregate`?

Reply:
It seems very weird to include service into aggregate.

Reply:
I don't see it as a "service" in DDD sense ... just a tiny component behind an abstract interface. Dunno, I don't feel strongly about this.